### PR TITLE
[Enhancement](DOE): Doe support object/nested use string

### LINKED ISSUE #12387: [Enhancement] Doe support query object/nested data
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Doe support query object/nested data

### Solution

object/nested data mapping to string value.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close https://github.com/apache/doris/issues/12387

## Problem summary

Doe support object/nested use string
1. doc_value logic modify, ignore object/nested
2. auto create table mapping object/nested to string type.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 1]
```diff
@@ -403,16 +404,15 @@ public static Type toDorisType(String esType) {
             case "double":
             case "scaled_float":
                 return Type.DOUBLE;
+            case "date":
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:407]: I think you should use ScalarType.getDefaultDateType(Type.DATE) to enable users to read date as datev2 type

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/MappingPhase.java [hunk 2]
```diff
@@ -119,6 +119,8 @@ private static void resolveDocValuesFields(SearchContext searchContext, JSONObje
                 if (!docValue) {
                     return;
                 }
+            } else if (fieldType == null) {
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/MappingPhase.java:122]: Add some comments to explain why we need this.

**Comment by stalary** [comment]
@qidaye @Gabriel39 done.

**Review by qidaye — DISMISSED** [review]
LGTM

**Review by Gabriel39 — DISMISSED** [review]
LGTM

File: be/src/http/http_client.cpp [hunk 3]
```diff
@@ -162,7 +163,8 @@ Status HttpClient::execute(const std::function<bool(const void* data, size_t len
     _callback = &callback;
     auto code = curl_easy_perform(_curl);
     if (code != CURLE_OK) {
-        LOG(WARNING) << "fail to execute HTTP client, errmsg=" << _to_errmsg(code);
+        LOG(WARNING) << "fail to execute HTTP client, errmsg=" << _to_errmsg(code)
+                     << " trace=" << get_stack_trace();
```
Comment (lide-reed) [inline be/src/http/http_client.cpp:167]: trace --> ,trace ?

**Review by lide-reed — DISMISSED** [review]
LGTM

File: be/src/http/http_client.cpp (hunk 3 above)
Comment (stalary) [reply be/src/http/http_client.cpp]: done.

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**